io.c

```diff
@@ -1114,6 +1114,12 @@ io_fflush(rb_io_t *fptr)
 int
 rb_io_wait_readable(int f)
 {
+    VALUE selector = rb_current_thread_selector();
+    if (selector != Qnil) {
+	VALUE result = rb_funcall(selector, rb_intern("wait_readable"), 1, INT2NUM(f));
```

Comment:
Should `wait_readable` receive the original `IO` object rather than a raw Integer?
I think that would be better and safer, but I'm not sure all callers of `rb_io_wait_readable` have access to the corresponding `IO` object.

Reply:
Just bearing in mind that the only reason to use `IO` in the test selector is because `IO.select` can't accept raw integers.

Reply:
Definitely pass the original IO. Raw numeric file descriptors are a problem (i.e. impossible) when JRuby runs in "pure Java" mode without native POSIX support. Stick to the proper abstractions; a file descriptor is just a number.

Reply:
As you can imagine, my hands are tied based on the C API that is used by C-exts. So, we may need to just support both.